File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < n; i++) {
            int r = target - nums[i];
            if (map.containsKey(r)) {
                return new int[]{map.get(r), i};
            }
            map.put(nums[i], i);
        }

        return new int[]{};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}
        n = len(nums)
    
        for i in range(n):
            numMap[nums[i]] = i

        for i in range(n):
            x = target - nums[i]
            if x in numMap and numMap[x] != i:
                return [i, numMap[x]]

        return []